### DISCUSSION TITLE
npm update option to preserve linked packages

### DISCUSSION THREAD

**Discussion by kevinoid** [npm/feedback#460] [CLI] [answered]
Suppose I have a package with a few dependencies, some installed from a registry, some from `npm link`:

```sh
npm init -y
npm install lodash
npm install commander
npm link commander
```

I would like to have a way to update the non-linked dependencies (`lodash`, but not `commander` in the above example) without replacing linked packages with versions from the package registry (e.g. because I'm hacking on them or testing with local modifications or they haven't been published yet).  `npm update` does not appear to provide such an option.  After running `npm update`, linked versions are always replaced with the latest version from the registry (or an error occurs if the version can't be satisfied).

Thanks for considering,
Kevin

**Comment by MylesBorins** [comment, marked as answer]
Hey @kevinoid the next step for this request would be to either open a new RFC to describe how this behavior could work in https://github.com/npm/rfcs or to create an issue in https://github.com/npm/cli if you believe this is more of a bug

**Reply by kevinoid** [reply]
Thanks @MylesBorins!  Good advice.  Sorry for my slow reply.

Apparently the issue has been discussed a few times over the years.  Most recently in npm/cli#2372 (which has links to previous discussions).  Not sure how I missed it.

If opening an RFC would help, I'd be willing to work on one.  Otherwise, I'll plan to just follow those.

Thanks again!
